### PR TITLE
Bluetooth: BASS: Add support for dynamic registration of BASS

### PR DESCRIPTION
**Pull request by babrsn**
Added support for dynamic registration and unregistration of the Broadcast Audio Scan Service (BASS) within the scan delegator. This enables both scan delegator and BASS to be registered or unregistered dynamically at runtime.

### CODE REVIEW

File: include/zephyr/bluetooth/audio/bap.h [hunk 1]
```diff
@@ -2042,14 +2042,28 @@ int bt_bap_broadcast_sink_delete(struct bt_bap_broadcast_sink *sink);
...
+ */
+int bt_bap_scan_delegator_register(struct bt_bap_scan_delegator_cb *cb);
+
+/**
+ * @brief unregister the Basic Audio Profile Scan Delegator and BASS.
+ *
+ * Unregister the scan deligator and Broadcast Audio Scan Service (BASS)
+ * dynamically at runtime.
+ *
+ * @return int   Error value. 0 on success, errno on fail.
```
Comment (fredrikdanebjer) [inline include/zephyr/bluetooth/audio/bap.h:2064]:
```suggestion
 * @return 0 in case of success or negative value in case of error.
```

Mainly for consistency across LE Audio.
I have to admit that I don't know if Zephyr dictates this, but I have not seen the return type attached after @return in this repo yet.

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 2]
```diff
@@ -1170,10 +1170,52 @@ BT_GATT_SERVICE_DEFINE(bass_svc,
...
+
+static bool scan_delegator_registered;
+static bool pa_sync_cb_registered;
+
+static int bt_bap_bass_register(void)
+{
+	int err;
+
+	if (bass_svc.attr_count == 0) {
+		LOG_ERR("BASS service not initialized properly");
```
Comment (fredrikdanebjer) [inline subsys/bluetooth/audio/bap_scan_delegator.c:1185]: Since this initialization happens statically, this shouldn't be able to happen. On this premise, IF this happens something is very wrong. Consider whether we should use an ASSERT here

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 3]
```diff
@@ -1186,17 +1228,28 @@ static int bt_bap_scan_delegator_init(void)
...
-	scan_delegator_cbs = cb;
+	if (!scan_delegator_registered) {
+		LOG_DBG("Scan delegator not yet registered\n");
+		return -EALREADY;
+	}
+
+	scan_delegator_cbs = NULL;
+	scan_delegator_registered = false;
+
+	return bt_bap_bass_unregister();
```
Comment (fredrikdanebjer) [inline subsys/bluetooth/audio/bap_scan_delegator.c:1252]: If `bt_bap_bass_unregister` fails we probably don't want to reset `scan_delegator_cbs` and `scan_delegator_registered`

File: include/zephyr/bluetooth/audio/bap.h (hunk 1 above)
Comment (babrsn) [reply include/zephyr/bluetooth/audio/bap.h]: OK. I just copied this line from the description of the function right after my added functions!
But anyway, I updated it.

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 2 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c]: Good point...

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 3 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c]: That sounds true, corrected.

File: include/zephyr/bluetooth/audio/bap.h (hunk 1 above)
Comment (fredrikdanebjer) [reply include/zephyr/bluetooth/audio/bap.h]: Intresting :) Anyway, lets be happy we got closer to consistency now :D

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 4]
```diff
@@ -1170,10 +1170,60 @@ BT_GATT_SERVICE_DEFINE(bass_svc,
...
+static struct bt_gatt_service bass_svc = BT_GATT_SERVICE(attr_bass_svc);
+
+static bool scan_delegator_registered;
+static bool pa_sync_cb_registered;
+
+static int bt_bap_bass_register(void)
+{
+	int err;
+
+	__ASSERT(bass_svc, "BASS GATT service is NULL!");
```
Comment (fredrikdanebjer) [inline subsys/bluetooth/audio/bap_scan_delegator.c:1184]: Since `bass_svc` is not a pointer this cannot fail. However, the same is true even if you do `&bass_svc`, as its lifetime always will surpass that of the funciton scope.

I think that we should check what you initially had, e.g. `bass_svc.attr_count != 0`. I think it would also be fine with just removing this check altogether though, since bass_svc never is un/reassigned. The advantage of keeping it would ofc. be that it will guard against future changes that reassigns bass_svc - I haven no idea how likely that would be. And perhaps such a potential change should have responsibility of asserting a good value too.

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 2 above)
Comment (fredrikdanebjer) [reply subsys/bluetooth/audio/bap_scan_delegator.c]: Since bass_svc is not a pointer this cannot fail. However, the same is true even if you do &bass_svc, as its lifetime always will surpass that of the funciton scope.

I think that we should check what you initially had, e.g. bass_svc.attr_count != 0. I think it would also be fine with just removing this check altogether though, since bass_svc never is un/reassigned. The advantage of keeping it would ofc. be that it will guard against future changes that reassigns bass_svc - I haven no idea how likely that would be. And perhaps such a potential change should have responsibility of asserting a good value too.

File: doc/releases/migration-guide-4.0.rst [hunk 5]
```diff
@@ -278,6 +278,16 @@ Bluetooth Audio
   :c:func:`bt_bap_unicast_server_unregister` has been called.
   (:github:`76632`)
 
+* The Broadcast Audio Scan Service (BASS) can now be registered and unregistered dynamically
```
Comment (Thalley) [inline doc/releases/migration-guide-4.0.rst:281]:
```suggestion
* The Broadcast Audio Scan Service (BASS) shall now be registered and unregistered dynamically
```

Comment (Thalley) [inline doc/releases/migration-guide-4.0.rst:281]: This should also mention that `bt_bap_scan_delegator_register_cb` has been removed/merged with `bt_bap_scan_delegator_register`

File: doc/releases/migration-guide-4.0.rst [hunk 6]
```diff
@@ -278,6 +278,16 @@ Bluetooth Audio
...
+  at runtime within the scan delegator. Two new APIs, :c:func:`bass_scan_delegator_register()`
+  and :c:func:`bass_scan_delegator_unregister()`, have been introduced to manage both BASS and
```
Comment (Thalley) [inline doc/releases/migration-guide-4.0.rst:283]:
```suggestion
  at runtime within the scan delegator. Two new APIs, :c:func:`bt_bap_scan_delegator_register()`
  and :c:func:`bt_bap_scan_delegator_unregister()`, have been introduced to manage both BASS and
```

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 2 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c]: True, if we need to check it, we can either do it as before or use __ASSERT for 'attr_bass_svc'. However, I agree with you that there’s no need to check it since it is initialized right before this function. So, I’ll remove it.

File: doc/releases/migration-guide-4.0.rst [hunk 7]
```diff
@@ -278,6 +278,16 @@ Bluetooth Audio
...
+  scan delegator registration and initialization dynamically.
+  This change allows more flexibility when registering or unregistering scan delegator and BASS
+  related functionality (e.g., Auracast) without requiring build-time configuration. Existing
```
Comment (Thalley) [inline doc/releases/migration-guide-4.0.rst:286]:
```suggestion
  This change allows more flexibility when registering or unregistering scan delegator and BASS
  related functionality without requiring build-time configuration. Existing
```

I don't think we should mention Auracast in BAP context; BAP is unaware of Auracast

File: doc/releases/migration-guide-4.0.rst [hunk 8]
```diff
@@ -278,6 +278,16 @@ Bluetooth Audio
...
+  applications that statically register BASS may need to be updated to use these new APIs if
+  dynamic control is required.
```
Comment (Thalley) [inline doc/releases/migration-guide-4.0.rst:288]:
```suggestion
  related functionality (e.g., Auracast) without requiring build-time configuration. Existing
  need to be updated to use these new APIs.
```

We should not have both dynamic and static registration.

File: include/zephyr/bluetooth/audio/bap.h [hunk 9]
```diff
@@ -2048,14 +2048,28 @@ int bt_bap_broadcast_sink_delete(struct bt_bap_broadcast_sink *sink);
...
  *
+ * Register the scan deligator and Broadcast Audio Scan Service (BASS)
+ * dynamically at runtime.
+ *
+ * @param cb Pointer to the callback struct
  * Only one set of callbacks can be registered at any one time, and calling this function multiple
  * times will override any previously registered callbacks.
  *
- * @param cb Pointer to the callback struct
+ * @return int   Error value. 0 on success, errno on fail.
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/bap.h]:
```suggestion
 * @return 0 on success, errno on fail.
```

File: include/zephyr/bluetooth/audio/bap.h [hunk 10]
```diff
@@ -2048,14 +2048,28 @@ int bt_bap_broadcast_sink_delete(struct bt_bap_broadcast_sink *sink);
...
 /**
- * @brief Register the callbacks for the Basic Audio Profile Scan Delegator
+ * @brief Register the Basic Audio Profile Scan Delegator and BASS.
  *
+ * Register the scan deligator and Broadcast Audio Scan Service (BASS)
+ * dynamically at runtime.
+ *
+ * @param cb Pointer to the callback struct
  * Only one set of callbacks can be registered at any one time, and calling this function multiple
  * times will override any previously registered callbacks.
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/bap.h:2058]:
```suggestion
 * Only one set of callbacks can be registered at any one time, and calling this function multiple
 * times will override any previously registered callbacks.
 * 
 * @param cb Pointer to the callback struct
```

File: subsys/bluetooth/audio/Kconfig.bap [hunk 11]
```diff
@@ -211,6 +211,7 @@ config BT_BAP_SCAN_DELEGATOR
 	select BT_EXT_ADV
 	select BT_PER_ADV_SYNC
 	select BT_ISO_SYNC_RECEIVER
+	select BT_GATT_DYNAMIC_DB
```
Comment (Thalley) [inline subsys/bluetooth/audio/Kconfig.bap:214]:
```suggestion
	depends on BT_GATT_DYNAMIC_DB
```

File: include/zephyr/bluetooth/audio/bap.h [hunk 12]
```diff
@@ -2048,14 +2048,28 @@ int bt_bap_broadcast_sink_delete(struct bt_bap_broadcast_sink *sink);
...
  *
- * @param cb Pointer to the callback struct
+ * @return int   Error value. 0 on success, errno on fail.
  */
-void bt_bap_scan_delegator_register_cb(struct bt_bap_scan_delegator_cb *cb);
+int bt_bap_scan_delegator_register(struct bt_bap_scan_delegator_cb *cb);
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/bap.h:2062]: Is there any future plans to add a register parameter to set e.g. the number of receive states?

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 13]
```diff
@@ -1170,10 +1170,58 @@ BT_GATT_SERVICE_DEFINE(bass_svc,
 	RECEIVE_STATE_CHARACTERISTIC(2)
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 2 */
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 1 */
-);
+};
+
+static struct bt_gatt_service bass_svc = BT_GATT_SERVICE(attr_bass_svc);
+
+static bool scan_delegator_registered;
+static bool pa_sync_cb_registered;
```
Comment (Thalley) [inline subsys/bluetooth/audio/bap_scan_delegator.c:1178]: Suggest to use `ATOMIC_DEFINE` for these instead so that we can better be thread safe.

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 14]
```diff
@@ -1170,10 +1170,58 @@ BT_GATT_SERVICE_DEFINE(bass_svc,
...
+		LOG_DBG("Failed to register BASS service (err %d)", err);
+		return err;
+	}
+
+	LOG_DBG("BASS service registered");
+
+	return 0;
+}
+
+static int bt_bap_bass_unregister(void)
```
Comment (Thalley) [inline subsys/bluetooth/audio/bap_scan_delegator.c:1195]:
```suggestion
static int bass_register(void)
{
	int err;

	err = bt_gatt_service_register(&bass_svc);
	if (err) {
		LOG_DBG("Failed to register BASS service (err %d)", err);
		return err;
	}

	LOG_DBG("BASS service registered");

	return 0;
}

static int bass_unregister(void)
```

Suggest to omit the `bt_bap` prefix for `static functions

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 15]
```diff
@@ -1186,17 +1234,35 @@ static int bt_bap_scan_delegator_init(void)
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 2 */
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 1 */
 
-	bt_le_per_adv_sync_cb_register(&pa_sync_cb);
+	if (!pa_sync_cb_registered) {
+		bt_le_per_adv_sync_cb_register(&pa_sync_cb);
```
Comment (Thalley) [inline subsys/bluetooth/audio/bap_scan_delegator.c:1238]: Missing error handling of `bt_le_per_adv_sync_cb_register`

File: subsys/bluetooth/audio/shell/bap_scan_delegator.c [hunk 16]
```diff
@@ -436,9 +436,15 @@ static int cmd_bap_scan_delegator_init(const struct shell *sh, size_t argc,
...
-		bt_bap_scan_delegator_register_cb(&scan_delegator_cb);
+		int err;
 
+		err = bt_bap_scan_delegator_register(&scan_delegator_cb);
+		if (err) {
+			shell_error(sh, "Failed to register scan delegator (err: %d)", err);
+			return -ENOEXEC;
+		}
+
+		bt_le_per_adv_sync_cb_register(&pa_sync_cb);
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/bap_scan_delegator.c:447]: Missing error handling of `bt_le_per_adv_sync_cb_register`

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 15 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:1238]: Is it really needed? because bt_le_per_adv_sync_cb_register only returns 0 (on success) and EEXIST (if already registered, which is not important or never happens here). I also can see that it hasn’t been handled in any other places…

Comment (Thalley) [reply subsys/bluetooth/audio/bap_scan_delegator.c:1238]: If it can only fail in the case of programming errors, then we should add an ASSERT IMO. 

The return value was recently added which would explain why it isn't being checked everywhere, but it should.

File: doc/releases/migration-guide-4.0.rst (hunk 5 above)
Comment (babrsn) [reply doc/releases/migration-guide-4.0.rst:281]: Done.

Comment (babrsn) [reply doc/releases/migration-guide-4.0.rst:281]: Done.

File: doc/releases/migration-guide-4.0.rst (hunk 6 above)
Comment (babrsn) [reply doc/releases/migration-guide-4.0.rst:283]: Done.

File: doc/releases/migration-guide-4.0.rst (hunk 7 above)
Comment (babrsn) [reply doc/releases/migration-guide-4.0.rst:286]: Ok.

File: doc/releases/migration-guide-4.0.rst (hunk 8 above)
Comment (babrsn) [reply doc/releases/migration-guide-4.0.rst:288]: True, corrected.

File: include/zephyr/bluetooth/audio/bap.h (hunk 9 above)
Comment (babrsn) [reply include/zephyr/bluetooth/audio/bap.h]: Done.

File: include/zephyr/bluetooth/audio/bap.h (hunk 10 above)
Comment (babrsn) [reply include/zephyr/bluetooth/audio/bap.h:2058]: Done.

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 13 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:1178]: Done.

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 14 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:1195]: Done.

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 15 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:1238]: Ok, fair enough. I’ll prefer to handle it by returning an error instead of using ASSERT, as more errors might be implemented in the future and the probable error is not fatal...

File: subsys/bluetooth/audio/shell/bap_scan_delegator.c (hunk 16 above)
Comment (babrsn) [reply subsys/bluetooth/audio/shell/bap_scan_delegator.c:447]: Done.

File: subsys/bluetooth/audio/Kconfig.bap (hunk 11 above)
Comment (Thalley) [reply subsys/bluetooth/audio/Kconfig.bap]: > Then I have to add it to all related configurations...

Yep :D 

But really we should always try to use `depends on` rather than `select` for various reasons: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#alternatives-to-select

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/Kconfig.bap (hunk 11 above)
Comment (babrsn) [reply subsys/bluetooth/audio/Kconfig.bap]: OK.

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 17]
```diff
@@ -1170,10 +1170,57 @@ BT_GATT_SERVICE_DEFINE(bass_svc,
 	RECEIVE_STATE_CHARACTERISTIC(2)
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 2 */
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 1 */
-);
+};
+
+static struct bt_gatt_service bass_svc = BT_GATT_SERVICE(attr_bass_svc);
+static ATOMIC_DEFINE(scan_delegator_registered, 1);
+static ATOMIC_DEFINE(pa_sync_cb_registered, 1);
```
[the remark this answers is not in the document]
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:1177]: OK.

File: include/zephyr/bluetooth/audio/bap.h (hunk 12 above)
Comment (Thalley) [reply include/zephyr/bluetooth/audio/bap.h:2063]: I agree. Better to do this over 2 PRs to keep it simpler

File: subsys/bluetooth/mesh/Kconfig [hunk 18]
```diff
@@ -37,7 +37,7 @@ config BT_MESH_GATT_CLIENT
 config BT_MESH_GATT_SERVER
 	bool
 	select BT_MESH_GATT
-	select BT_GATT_DYNAMIC_DB
+	depends on BT_GATT_DYNAMIC_DB
```
Comment (Thalley) [inline subsys/bluetooth/mesh/Kconfig:40]: If the Kconfig circular dependencies requires changes to everywhere, including mesh, then it should be fixed in a separate commit or maybe even a separate PR. Doing it together with the BASS changes is messy

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 19]
```diff
@@ -87,7 +87,16 @@ struct bt_bap_scan_delegator_inst {
 		[CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT];
 };
 
-static bool conn_cb_registered;
+enum scan_delegator_registered_flags {
+	REGISTERED_CONN_CB,
+	REGISTERED_SCAN_DELIGATOR,
+	REGISTERED_PA_SYNC_CB,
+
+	REGISTERED_FLAGS_NUM,
```
Comment (Thalley) [inline subsys/bluetooth/audio/bap_scan_delegator.c:95]: I strongly prefer that the enum name is prefix in the values:
```suggestion
enum scan_delegator_registered_flags {
	SCAN_DELEGATOR_REGISTERED_CONN_CB,
	SCAN_DELEGATOR_REGISTERED_SCAN_DELIGATOR,
	SCAN_DELEGATOR_REGISTERED_PA_SYNC_CB,

	SCAN_DELEGATOR_REGISTERED_FLAGS_NUM,
```

Consider also just renaming this to `scan_delegator_flag` so that it can be used for non-register things in the future

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 20]
```diff
@@ -325,9 +334,9 @@ static struct broadcast_assistant *get_bap_broadcast_assistant(struct bt_conn *c
 		}
 	}
 
-	if (!conn_cb_registered) {
+	if (!atomic_test_bit(registered_flags, REGISTERED_CONN_CB)) {
 		bt_conn_cb_register(&conn_cb);
-		conn_cb_registered = true;
+		atomic_set_bit(registered_flags, REGISTERED_CONN_CB);
```
Comment (Thalley) [inline subsys/bluetooth/audio/bap_scan_delegator.c:339]: Please have another look at `aics_client`. 

It's not obvious, but it's quite important that we use `atomic_test_and_set_bit` instead of `atomic_test_bit` + `atomic_set_bit`, as the former is thread safe and the latter is not. 

So basically it's something like


```
if (atomic_test_and_set_bit(...)) { abort; }

if (err) {atomic_clear_bit(...))
```

That way we take control of the bit immediately and prevent other threads from passing with `atomic_test_bit` and we can release it in case of an error.

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/Kconfig (hunk 18 above)
Comment (babrsn) [reply subsys/bluetooth/mesh/Kconfig:40]: Then what’s your suggestion for this PR? Should I revert to using ´select´ (instead of ´depends on´) temporarily for ´BT_GATT_DYNAMIC_DB´ in ´BT_BAP_SCAN_DELEGATOR´ definition for this PR and then fix it in another PR?

Comment (Thalley) [reply subsys/bluetooth/mesh/Kconfig:40]: We can

1. Revert to using `select` and fix in another PR
2. Add a commit preceding this that fixes the dependency issues and then add the BASS commit on top of that
3. Create new PR fixing the dependency issues and then put this PR on top of that PR (creating 2 PRs). 

I don't have a strong preference on which direction to take, so will leave that up to you

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 20 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:339]: Ok, now I get your point :)

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 19 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:95]: Ok.

File: subsys/bluetooth/mesh/Kconfig (hunk 18 above)
Comment (babrsn) [reply subsys/bluetooth/mesh/Kconfig:40]: Ok, I prefer the first option :)
So, I'll update the PR accordingly...

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 21]
```diff
@@ -87,7 +87,16 @@ struct bt_bap_scan_delegator_inst {
...
-static bool conn_cb_registered;
+enum scan_delegator_registered_flags {
+	SCAN_DELEGATOR_REGISTERED_CONN_CB,
+	SCAN_DELEGATOR_REGISTERED_SCAN_DELIGATOR,
+	SCAN_DELEGATOR_REGISTERED_PA_SYNC_CB,
+
+	SCAN_DELEGATOR_REGISTERED_FLAGS_NUM,
+};
+
+static ATOMIC_DEFINE(registered_flags, SCAN_DELEGATOR_REGISTERED_FLAGS_NUM);
```
Comment (Thalley) [inline subsys/bluetooth/audio/bap_scan_delegator.c:98]:
```suggestion
enum scan_delegator_flag {
	SCAN_DELEGATOR_FLAG_REGISTERED_CONN_CB,
	SCAN_DELEGATOR_FLAG_REGISTERED_SCAN_DELIGATOR,
	SCAN_DELEGATOR_FLAG_REGISTERED_PA_SYNC_CB,

	SCAN_DELEGATOR_FLAG_NUM,
};

static ATOMIC_DEFINE(flags, SCAN_DELEGATOR_FLAG_NUM);
```

Might have missed my suggestion in previous review: Suggest to keep the enum name more generic so that it can be expanded to non-register flags as well :)

**Review by Thalley — COMMENTED** [review]
Suggestion on some naming, otherwise LGTM

File: subsys/bluetooth/audio/bap_scan_delegator.c (hunk 21 above)
Comment (babrsn) [reply subsys/bluetooth/audio/bap_scan_delegator.c:98]: uh, ok ;)

**Review by Thalley — DISMISSED** [review]
thanks for the feature and the quick updates :)

**Review by niym-ot — DISMISSED** [review]

**Review by Thalley — APPROVED** [review]

**Review by niym-ot — APPROVED** [review]
